flow/connectors/clickhouse/clickhouse.go

```diff
@@ -29,11 +29,12 @@ import (
 
 type ClickHouseConnector struct {
 	*metadataStore.PostgresMetadata
-	database      clickhouse.Conn
-	logger        log.Logger
-	config        *protos.ClickhouseConfig
-	credsProvider *utils.ClickHouseS3Credentials
-	s3Stage       *ClickHouseS3Stage
+	database           clickhouse.Conn
+	dbSupportsExchange bool
```

Comment:
this won't persist between validation & rename

Reply:
this needs to be populated by the `NewClickHouseConnector` method instead

can be populated by just running `SELECT engine_full FROM system.databases` unless EXCHANGE needs extra permissions